backend/dataInterface/mongoDB.js

```diff
@@ -79,8 +100,181 @@ module.exports.getFireCommentsByFire = async (id) => {
   } finally {
     await client.close();
   }
+};
+
+//create a comment by fireid, userid
+module.exports.createComment = async (paramObj, newObj) => {
+  try {
+    await client.connect();
+    const db = client.db(databaseName);
+    const collection = db.collection(comments_Collection);
+    const commentObj = {
+      ...newObj,
+      fire_id: ObjectId(paramObj.id),
+      username: paramObj.userName,
+      text: newObj.text,
+      createdDate: new Date(),
+    };
+    const result = await collection.insertOne(commentObj);
+    if (result.acknowledged) {
+      return {
+        newObjectId: result.insertedId,
+        message: `Comment created! ID: ${result.insertedId}`,
+      };
+    } else {
+      return { error: "New comment insert failed." };
+    }
+  } catch (err) {
+    console.log(err);
+  } finally {
+    await client.close();
+  }
+};
+//update comment by comment id
+module.exports.updateComment = async (commentId, newObj) => {
+  try {
+    await client.connect();
+    const db = client.db(databaseName);
+    const collection = db.collection(comments_Collection);
+    const updateRules = {
+      $set: { text: newObj.text },
+    };
+    const filter = { _id: ObjectId(commentId) };
+    const result = await collection.updateOne(filter, updateRules);
+
+    if (result.modifiedCount != 1) {
+      return {
+        error: `Something went wrong. ${result.modifiedCount} comment was not updated. Please try again.`,
+      };
+    }
+
+    const updatedComment = module.exports.getFireCommentByCommentId(commentId);
+    return updatedComment;
+  } catch (err) {
+    return { Error: "Failed to Update Comment" };
+  } finally {
+    await client.close();
+  }
+};
+//delete comment by comment id
+module.exports.deleteComment = async (commentId) => {
+  try {
+    await client.connect();
+    const db = client.db(databaseName);
+    const collection = db.collection(comments_Collection);
+
+    const deletionRules = { _id: ObjectId(commentId) };
+    const result = await collection.deleteOne(deletionRules);
+    if (result.deletedCount != 1) {
+      return {
+        error: `Something went wrong. ${result.deletedCount} comments were not deleted. Please try again.`,
```

Comment:
Same comment as above. `result.deletedCount` is the count of items that were successfully deleted.